include/handler.h

```diff
@@ -456,17 +463,24 @@ class handler {
 		// Note that cgf_diagnostics has a similar check, but we don't catch void side effects there.
 		if(!m_side_effects.empty()) { throw std::runtime_error{"Side effects cannot be used in device kernels"}; }
 		m_task =
-		    detail::task::make_device_compute(m_tid, geometry, std::move(launcher), std::move(m_access_map), std::move(m_reductions), std::move(debug_name));
+		    detail::task::make_device_compute(m_tid, geometry, std::move(launcher), std::move(m_access_map), std::move(m_reductions));
+
+		m_task->set_debug_name(m_debug_task_name.value_or(debug_name));
```

Comment:
Can we improve naming of the `debug_name` parameter or `m_debug_task_name` to emphasize that one is from the kernel name template parameter and one is from `set_task_name`? This line is a bit confusing otherwise

Reply:
should be better now